tiledb/common/thread_pool/thread_pool.cc

```diff
@@ -30,28 +30,34 @@
  * This file defines the ThreadPool class.
  */
 
-#ifdef LEGACY_THREAD_POOL
-#include "legacy_thread_pool.cc"
-#else
-
 #include <cassert>
 #include <memory>
 #include <queue>
+#include <thread>
 
 #include "tiledb/common/logger.h"
 #include "tiledb/common/thread_pool.h"
 
 namespace tiledb::common {
 
-Status ThreadPool::init(size_t n) {
-  if (n == 0) {
-    return Status_ThreadPoolError(
-        "Unable to initialize a thread pool with a concurrency level of 0.");
+ThreadPool::ThreadPool(size_t n)
+    : concurrency_level_(n) {
+  // If concurrency_level_ is set to zero, shut down the thread pool
```

Comment:
Why would we have tasks to drain in the constructor?

Reply:
"Shut down" isn't the same as "has no current tasks". "Shut down" means "has no current tasks" and "not accepting new tasks". It's the second of those properties that this comment is about.

Reply:
Marking resolved, with a more specific question below.